File: src/main/kotlin/com/sainnt/netshop/entity/ProductDescription_.java
package com.sainnt.netshop.entity;


import javax.annotation.Generated;
import javax.persistence.metamodel.SingularAttribute;
import javax.persistence.metamodel.StaticMetamodel;

@Generated(value = "org.hibernate.jpamodelgen.JPAMetaModelEntityProcessor")
@StaticMetamodel(ProductDescription.class)
public abstract class ProductDescription_ {
    public static volatile SingularAttribute<ProductDescription, Integer> id;
    public static volatile SingularAttribute<ProductDescription, String> description;
    public static volatile SingularAttribute<ProductDescription, Product> product;
}

File: src/main/kotlin/com/sainnt/netshop/entity/ProductDescription.kt
package com.sainnt.netshop.entity

import javax.persistence.*

@Entity
@Table(name = "product_description")
@SequenceGenerator(
    name = "productDescriptionSequence",
    sequenceName = "PRODUCT_DESCRIPTION_SEQ",
    initialValue = 1,
    allocationSize = 1
)
class ProductDescription(
    @Column(nullable = false)
    var description: String,
    @OneToOne
    @JoinColumn(name = "product_id")
    val product: Product
) {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "productDescriptionSequence")
    var id: Long? = null
}